src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -19,12 +23,37 @@ public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNu
             ColNumber = colNumber;
         }
     }
-
+    
     public interface INodeWithDebugInfo
     {
         DebugLocInfo[] DebugLocInfos
         {
             get;
         }
+
+        DebugVarInfo[] DebugVarInfos
+        {
+            get;
+        }
+    }
+
+     
+    public struct DebugVarInfo
+    {
+        public readonly uint VarNumber; // For parameters, this is the parameter index.
+                                        // For local variables, this is the slot number.
+        public readonly string Name;
+        public readonly uint TypeIndex;
+        public readonly bool IsParam;
+        public List<NativeVarInfo> Ranges;
```

Comment:
Thanks for pointing this out. However, according to the usage of the field in other parts of this change where copies and assignments are involved, I'd like to keep it as it is now since ArrayBuilder is a struct type with an array field. Changing among ArrayBuilder copies may violate the array field.